alphagov/tdt-documentation
# Re-do of overhaul branch

Created new overhaul branch from master to resolve conflicts with existing overhaul branch that was created in November 2019. Squashed following commits:

Redo create a project content

revised structure commit 1

revised structure commit 2

revised structure commit 3

4th commit

Remove 'Review project section'

Removing section about reviewing because it contains GDS-specific
process guidance. This documentation is about getting people to use the
tech docs template, not about process around documentation.

In more detail, the sections removed are:

- 'Governance' - not strictly to do with the template, but with
processes around  docs and the team
- '2i review' - a process thing we do at GDS, not to do with the
template
- 'Submit a pull request' - again, how we do it at GDS. With the
toolchain being docs-as-code, changes may not happen via PRs

"Maintain..." section to include Daniel the Manual Spaniel

Regular reviews of documentation content are a part of maintenance.

This commit:

- moves content about Daniel to the Maintenance section
- adds minimal viable content about how Daniel works and how to set it
up (link to README)
- addds a screenshot to show what the notification looks like
- moves all link addresses Daniel uses to a partial which defines link
labels.
- updates page  expiry frontmatter

Rename `update_gem` to `use_latest_template` and improve content

- change page title to less technical terminology because the audience
are likely not necessarily familiar with Ruby
- remove section about cloning the repository because it's not a
necessary step and users very likely already have the repository. This
documentation is not for explaining Git.
- explain that latest template means latest gem
- give example command and example output
- update review date in the frontmatter

Write content for setting up redirects

* explain why one should set up redirect
* give example of when one should set up redirects
* link to places that explain how to set up redirects in all 3 ways
* update page expiry in frontmatter

Use `<%%` so ERB doesn't evaluate the code block examples

There was a random "You can put anything in the sidebar" at the top of
the sidebar for the Frontmatter page...haha

Add overview content for "Mainain..." section

* explain what the section is for
* link to the pages in this section
* create new link labels, as needed
* update review date

Make #tech-writers the owner for the Tech Docs Template documentation

GDS technical writers look after the docs for the docs template so their
channel should appear as the owner.

Docs Template related websites.

Tidy up URLs and page titles

- do not use abbreviated words in content folder names beause they
become part of the URL and we're not meant to abbreviate in URLs

* remove double quotes around titles in the frontmatter

Put all links in a separate links partial

- the partial containing the link label definitions is read in at the
bottom of every page
- single source of truth for all links
- makes content more readable

changes after AS review

Combined single and multipage content into one Structure your Documentation section. Write Build your documentation section. Write version control section. Update Edit your content section.

Update gemfiles to master branch equivalent

fix broken links

fix redirects

remove layout images in frontmatter

Add alt attributes to banner images

fix images hopefully

remove references to incomplete content and add to github issues

accessibility fixes for links

Accessibility fixes

Changes post content crit

Fix broken partials

Fix tech-docs.yml

pre-i changes

changes post pre-i

fix broken link

fix typos

changes post-2i

### Context


### Changes proposed in this pull request


### Guidance to review
